corehq/apps/sso/static/sso/js/enterprise_edit_identity_provider.js

```diff
@@ -55,6 +56,41 @@ hqDefine('sso/js/enterprise_edit_identity_provider', [
 
         };
 
+        let remoteUserManagementAPISecretManager = function () {
+            let self = {};
+            self.isCancelUpdateVisible = ko.observable(false);
+            self.apiExpirationDate = "";
+
+            self.dateApiSecretExpiration = ko.observable($('#id_date_api_secret_expiration').val());
+            self.isAPISecretVisible =  ko.observable(!initialPageData.get('api_secret_exists'));
+            self.apiSecret = ko.observable();
+
+            self.showAPISecret = function () {
+                self.isAPISecretVisible(true);
+                self.isCancelUpdateVisible(true);
+                // Store the current expiration date before clearing them for editing.
+                self.apiExpirationDate = self.dateApiSecretExpiration();
+                self.dateApiSecretExpiration('');
+            };
+
+            self.hideAPISecret = function () {
+                self.isAPISecretVisible(false);
+                self.isCancelUpdateVisible(false);
+                // Reset the api secret to blank if user cancel editing
+                self.apiSecret('');
+                // Restore the original values of expiration date after canceling editing.
+                self.dateApiSecretExpiration(self.apiExpirationDate);
+
+            };
+            return self;
+
+        };
+
+        if (initialPageData.get('toggle_api_secret')) {
```

Comment:
Hi Matt, although they are never simultaneously `true`, but they can be simultaneously `false`. So I don't think `if/else` will work?

Reply:
@mjriley they can be simultaneously `false`. However, the view model for OIDC and for SAML will need to be merged anyway with the introduction of the API key management changes you are making. I don't think this should block Jing's PR now, however, as this PR has been open for your review for a while. It would be best to first merge these changes from Jing's PR and you can merge the two view models in the API Key changes. :)

Reply:
@mjriley I understand your concerns about two view models being applied to the same HTML. This will result in an error if this is the case. I think the solution moving forward is to merge these two view models, especially since remote user management will be available for both SAML and OIDC.

Reply:
@mjriley @jingcheng16 ok plan is to sync tomorrow during the dev block and pair on how we merge these models so that both PRs can move forward! Let's also work to find a solution about the naming. Thanks!